File: 2d a 1d columna/columna.cpp
#include <iostream>
using namespace std;

int main() {
    int matriz2D[3][3] = {{1,2,3},{4,5,6},{7,8,9}};
    int matriz1D[9], index = 0;

    cout << "Matriz 2D original:\n";
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            cout << matriz2D[i][j] << " ";
        }
        cout << endl;
    }

    for (int j = 0; j < 3; j++) {
        for (int i = 0; i < 3; i++) {
            matriz1D[index++] = matriz2D[i][j];
        }
    }

    cout << "\nMatriz 1D (por columnas):\n";
    for (int i = 0; i < 9; i++)
        cout << matriz1D[i] << " ";

 
}

File: 2d a 1d columna/columna.java
public class Main {
    public static void main(String[] args) {
        int[][] matriz2D = {{1,2,3},{4,5,6},{7,8,9}};
        int[] matriz1D = new int[9];
        int index = 0;

        System.out.println("Matriz 2D original:");
        for (int i = 0; i < matriz2D.length; i++) {
            for (int j = 0; j < matriz2D[i].length; j++) {
                System.out.print(matriz2D[i][j] + " ");
            }
            System.out.println();
        }

        for (int j = 0; j < matriz2D[0].length; j++) {
            for (int i = 0; i < matriz2D.length; i++) {
                matriz1D[index++] = matriz2D[i][j];
            }
        }

        System.out.println("\nMatriz 1D (por columnas):");
        for (int val : matriz1D)
            System.out.print(val + " ");

       
    }
}